packages/blocks/src/api/utils.js

```diff
@@ -39,21 +38,16 @@ const ICON_COLORS = [ '#191e23', '#f8f9f9' ];
  * @return {boolean} Whether the block is an unmodified block.
  */
 export function isUnmodifiedBlock( block ) {
-	// Cache a created default block if no cache exists or the default block
-	// name changed.
-	if ( ! isUnmodifiedBlock[ block.name ] ) {
-		isUnmodifiedBlock[ block.name ] = createBlock( block.name );
-	}
-
-	const newBlock = isUnmodifiedBlock[ block.name ];
-	const blockType = getBlockType( block.name );
-
-	function isEqual( a, b ) {
-		return ( a?.valueOf() ?? a ) === ( b?.valueOf() ?? b );
-	}
+	return Object.entries( getBlockType( block.name )?.attributes ?? {} ).every(
```

Comment:
we're checking the attributes on the block type here, but would it be important to compare also the attributes on this particular block instance that may not be on the block type?

for instance, if I `setAttribute('formerlyNotPresentAttribute', true)` then I would imagine I have a modified block, but the block type won't have `formerlyNotPresentAttribute` so it won't show different here

Reply:
We didn't before. And I guess if it's not officially declared, why check?

Reply:
I missed that in the before code. makes me wonder anyway if it's been a latent bug. no need to make it part of this change.